swagger/swagger-invocation/invocation-core/src/main/java/org/apache/servicecomb/swagger/engine/SwaggerEnvironment.java

```diff
@@ -17,115 +17,78 @@
-    this.producerArgumentsFactory = producerArgumentsFactory;
-  }
-
-  public ConsumerArgumentsMapperFactory getConsumerArgumentsFactory() {
-    return consumerArgumentsFactory;
-  }
-
-  public void setConsumerArgumentsFactory(ConsumerArgumentsMapperFactory consumerArgumentsFactory) {
-    this.consumerArgumentsFactory = consumerArgumentsFactory;
-  }
-
-  public SwaggerConsumer createConsumer(Class<?> consumerIntf) {
-    // consumer interface equals to contract interface
-    return createConsumer(consumerIntf, consumerIntf);
-  }
+  public SwaggerConsumer createConsumer(Class<?> consumerIntf, Swagger swagger) {
+    Map<Class<?>, ContextArgumentMapperFactory> contextFactorys = SPIServiceUtils
+        .getOrLoadSortedService(ConsumerContextArgumentMapperFactory.class)
+        .stream()
+        .collect(Collectors.toMap(ConsumerContextArgumentMapperFactory::getContextClass, Function.identity()));
+    SwaggerOperations swaggerOperations = new SwaggerOperations(swagger);
 
-  public SwaggerConsumer createConsumer(Class<?> consumerIntf, Class<?> swaggerIntf) {
     SwaggerConsumer consumer = new SwaggerConsumer();
     consumer.setConsumerIntf(consumerIntf);
-    consumer.setSwaggerIntf(swaggerIntf);
     for (Method consumerMethod : consumerIntf.getMethods()) {
-      String swaggerMethodName = findSwaggerMethodName(consumerMethod);
-      // consumer参数不一定等于swagger参数
-      Method swaggerMethod = ReflectUtils.findMethod(swaggerIntf, swaggerMethodName);
-      if (swaggerMethod == null) {
-        // consumer method set bigger than contract, it's invalid
+      String operationId = findOperationId(consumerMethod);
+      SwaggerOperation swaggerOperation = swaggerOperations.findOperation(operationId);
+      if (swaggerOperation == null) {
+        // consumer method set is bigger than contract, it's invalid
         // but we need to support consumer upgrade before producer, so only log and ignore it.
         LOGGER.warn("consumer method {}:{} not exist in contract.",
             consumerIntf.getName(),
             consumerMethod.getName());
         continue;
       }
 
-      ArgumentsMapperConfig config = new ArgumentsMapperConfig();
-      config.setSwaggerMethod(swaggerMethod);
-      config.setProviderMethod(consumerMethod);
-
-      ConsumerArgumentsMapper argsMapper =
-          consumerArgumentsFactory.createArgumentsMapper(config);
-      ConsumerResponseMapper responseMapper = consumerResponseMapperFactorys.createResponseMapper(
-          swaggerMethod.getGenericReturnType(),
-          consumerMethod.getGenericReturnType());
+      ConsumerArgumentsMapperCreator creator = new ConsumerArgumentsMapperCreator(
+          Json.mapper().getSerializationConfig(),
+          contextFactorys,
+          consumerMethod,
+          swaggerOperation);
+      ConsumerArgumentsMapper argsMapper = creator.createArgumentsMapper();
+      ConsumerResponseMapper responseMapper = null;
```

Comment:
add a TODO?

Reply:
not important?
because not finish this feature, can not do remote invoke
and alreasy have related jira task: https://issues.apache.org/jira/browse/SCB-1212